### ISSUE TITLE
Experiment traffic medium & source is not being attributed

### ISSUE THREAD

**Issue by amychurchwell** [mozilla/bedrock#9126]
## Description

A recent A/B test on the firefox/new page shows that there seems to be an issue with traffic medium & source values not being passed.

Experiment data by medium & source. About **76%** of the data is attributed as `(none)`
![image](https://user-images.githubusercontent.com/30483988/86957091-0c1b3000-c10f-11ea-851f-5812d56688d7.png)

This is unusual because the split of traffic by medium on /new is normally like so. With **~27.1%** traffic attributed as `(none)`
![image](https://user-images.githubusercontent.com/30483988/86957100-0faeb700-c10f-11ea-8b20-aa873b306b45.png)

From slack:

> We also looked at a previous traffic cop experiment we ran in April and the same problem happened at that point too. Most of the traffic was being labeled as medium = (none) — when we know that if correct randomization/attribution had happened with traffic cop, we would have attribution splits that closely matched the splits from the live, non-experimental page. We’re looking to debug any potential issues that may be occurring with maintaining/passing down medium/source attribution. Any insights into why this might be happening would be really appreciated too!

[Traffic cop Github repo](https://github.com/mozilla/trafficcop/)
[Traffic cop docs on maintaining referral sources](https://github.com/mozilla/trafficcop/blob/master/documentation.md#maintaining-referral-sources)

**Comment by alexgibson** [comment]
@amychurchwell as you linked to [here](https://github.com/mozilla/trafficcop/blob/master/documentation.md#maintaining-referral-sources), Traffic Cop will (by natural effects of performing a redirect) appear as `(direct)` in GA. 

I took a look under the hood in bedrock, as it looks like we have some existing code in our GA implementation to handle passing the `customReferrer` to GTM: https://github.com/mozilla/bedrock/blob/master/media/js/base/core-datalayer-page-id.js#L47-L54

The TL:DR is: this referral data should be in GA, as it gets passed along in the `page-id-loaded` event. My guess is some extra GA reporting efforts might be needed to pull this data out and piece it together.

If this type of referral data is needed for an experiment, I think it may be a good idea in the future for the CRO team to make a QA checklist (of sorts), which can be used to verify experiment analytics on a demo server, prior to an experiment being put in the wild?

**Comment by rraue** [comment]
Does anyone know how to get this customReferrer in GA, in which view I can see it? @hoosteeno maybe?

**Comment by amychurchwell** [comment]
@alexgibson 

> If this type of referral data is needed for an experiment, I think it may be a good idea in the future for the CRO team to make a QA checklist (of sorts), which can be used to verify experiment analytics on a demo server, prior to an experiment being put in the wild?

Thank you for the suggestion! I will do this, but I want to mention that this is a North American Brand team experiment outside of the CRO team. Perhaps the checklist can live in bedrock docs.

**Comment by hoosteeno** [comment]
I have asked Adswerve to help debug. It kind of looks to me like GTM has stopped passing through the variable Alex mentions above.

**Comment by hoosteeno** [comment]
Adswerve will debug the data collection here, but we won't be able to manufacture historical data for this experiment. We can either wait for them to debug and then run the experiment again, or assume (maybe with data science help) that the enrolled visitors generally match our overall traffic patterns.

**Comment by alexgibson** [comment]
I believe this turned out to be a bug in our GA implementation as opposed to Traffic Cop, and has been resolved by our agency. Closing this, but feel free to re-open if still needed.
